File: Python/1614.py
class Solution:
    def maxDepth(self, s: str) -> int:
        cur = ans = 0
        for i in s:
            if i == '(':
                cur += 1
                ans = max(cur, ans)
            elif i == ')': cur -= 1
        
        return ans

# Topic: String
# Time Complexity: O(n)
# Space Complexity: O(1)

File: C++/1614.cpp
#include <string>
#include <algorithm>

class Solution {
public:
    int maxDepth(std::string s) {
        int cur = 0, ans = 0;
        for(auto i : s){
            if(i == '(') ans = std::max(ans, ++cur);
            else if(i == ')') --cur;
        }
        return ans;
    }
};

// Topic: String
// Time Complexity: O(n)
// Space Complexity: O(1)